doc/tutorials/computing-product-analytics.md

```diff
@@ -0,0 +1,258 @@
+---
+jupytext:
+  notebook_metadata_filter: myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.5
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Conduct Product Analysis Using SQL
+
++++
+
+Product analytics is a common analysis. This tutorial will provide a scenario for an e-commerce website and shows you how to conduct product analysis using SQL.
+
++++
+
+## Growth and Retention
+
++++
+
+Growth rate and user retention are some common metrics used in product analytics. Suppose that we have devleoped an app for Ploomber and want to see whether the app is popular. Here is some data we collected.
+
++++
+
+### Dataset Description
+
+For simplicity, we just use some small-scale randomly generated data. The dataset **user_activity** is consists of three columns: 
+- User_id : the unique identifier of the user
+- Month: the month where 
+- Activity_count: the number of activity conduct by the user in that month. If the user never used this app before this month, this is considered as their sign-up month. 
+
+
+| user_id | date       | activity_count |
+|--------|------------|----------------|
+| 1      | 2021-01-01 | 5              |
+| 1      | 2021-02-01 | 3              |
+| 1      | 2021-03-01 | 2              |
+| 2      | 2021-01-01 | 10             |
+| 3      | 2021-02-01 | 1              |
+| 3      | 2021-03-01 | 0              |
+| 4      | 2021-02-01 | 6              |
+| 5      | 2021-01-01 | 4              |
+| 5      | 2021-02-01 | 5              |
+| 5      | 2021-03-01 | 6              |
+| 6      | 2021-03-01 | 7              |
+| 7      | 2021-03-01 | 10             |
+
+```{code-cell} ipython3
+:tags: [hide-output]
+
+%pip install jupysql duckdb-engine --quiet
+```
+
+```{code-cell} ipython3
+%load_ext sql
+%sql duckdb://
+```
+
+Let's create the table by ourself first using SQL.
+
+```{code-cell} ipython3
+:tags: [hide-output]
+
+%%sql
+DROP TABLE IF EXISTS user_activity;
+CREATE TABLE user_activity (
+  user_id INT NOT NULL,
+ date DATE NOT NULL,
+  activity_count INT NOT NULL,
+  PRIMARY KEY (user_id, date)
+);
+INSERT INTO user_activity (user_id, date, activity_count)
+VALUES
+  (1, '2021-01-01', 5),
+  (1, '2021-02-01', 3),
+  (1, '2021-03-01', 2),
+  (2, '2021-01-01', 10),
+  (3, '2021-02-01', 1),
+  (3, '2021-03-01', 0),
+  (4, '2021-02-01', 6),
+  (5, '2021-01-01', 4),
+  (5, '2021-02-01', 5),
+  (5, '2021-03-01', 6),
+  (6, '2021-03-01', 7),
+  (7, '2021-03-01', 10);
+```
+
+Let's check our table using select.
+
+```{code-cell} ipython3
+:tags: [hide-output]
+
+%%sql
+SELECT * FROM user_activity
+```
+
+### Growth 
+
+We defined growth rate to be the percentage increase of total number of users between each month. 
```

Comment:
Past and present tense are mixed in the voice of the writer. Present tense appears more often throughout the tutorial. Suggest to pick present tense voice.

I like that you summarized what the code below does. Suggest to word it along the lines of

"Now that we have the growth rate as percentage increase of total number of users between each month, we can ..."